File: solutions/go/prime-factors/1/prime_factors.go
package prime

func Factors(number int64) (result []int64) {
	factor := int64(2)
	for number > 1 {
		if number%factor == 0 {
			result = append(result, factor)
			number /= factor
		} else {
			factor += 1
		}
	}
	return
}

File: solutions/javascript/prime-factors/1/prime-factors.js
/**
 * Function to compute the prime factors of a given number.
 * @param {number} n - The number to factorize.
 * @returns {number[]} An array of prime factors.
 * @example
 * primeFactors(28); // returns [2, 2, 7]
 */
export const primeFactors = (n) => {
  if (n < 2) return []; // No prime factors for numbers less than 2

  let num = n;
  const factors = [];
  let currentFactor = 2;

  while (num !== 1) {
    if (num % currentFactor === 0) {
      factors.push(currentFactor);
      num /= currentFactor;
    } else {
      currentFactor += 1;
    }
  }

  return factors;
};
